apps/builder/app/builder/features/style-panel/sections/transitions/transition-content.tsx

```diff
@@ -0,0 +1,75 @@
+import type { InvalidValue, LayersValue } from "@webstudio-is/css-engine";
+import {
+  Flex,
+  Label,
+  TextArea,
+  theme,
+  textVariants,
+} from "@webstudio-is/design-system";
+import { useState } from "react";
+import { parseTransition } from "@webstudio-is/css-data";
+import type { CreateBatchUpdate } from "../../shared/use-style-data";
+import type { IntermediateStyleValue } from "../../shared/css-value-input";
+
+type TransitionContentProps = {
+  index: number;
+  value: string;
+  onEditLayer: (index: number, layer: LayersValue) => void;
+  createBatchUpdate: CreateBatchUpdate;
+};
+
+export const TransitionContent = (props: TransitionContentProps) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    IntermediateStyleValue | InvalidValue | undefined
+  >();
+
+  const handleChange = (value: string) => {
+    setIntermediateValue({
+      type: "intermediate",
+      value,
+    });
+  };
+
+  const handleComplete = () => {
+    if (intermediateValue === undefined) {
+      return;
+    }
+    const layers = parseTransition(intermediateValue.value);
+    if (layers.type === "invalid") {
+      setIntermediateValue({
+        type: "invalid",
+        value: intermediateValue.value,
+      });
+      return;
+    }
+
+    props.onEditLayer(props.index, layers);
```

Comment:
I think we need to reset intermediate value to undefined in handleComplete. Imagine you have edited it, then AI will change store value, but in control you will always see the local state